src/bsp/src/stm32/src/interloc/src/states/src/WaitResponseState.cpp

```diff
@@ -8,25 +7,49 @@ WaitResponseState::WaitResponseState(ILogger& logger, DecawaveArray& decawaves)
     AbstractInterlocState(logger, decawaves) {}
 
 void WaitResponseState::process(InterlocStateHandler& context) {
-    m_decawaves[DecawavePort::A].receive(m_rxFrame, InterlocTimeManager::getResponseTimeout());
+    uint64_t pollTxTs = 0;
+    // record the actual time at wich the Poll was transmitted
+    m_decawaves[DecawavePort::A].getTxTimestamp(&pollTxTs);
```

Comment:
You can use the pollTxTs in the TWR structure instead. It will save an SPI operation

Reply:
done